File: fix_keywords.py
import os
import re

def fix_keywords_in_file(file_path):
    """将traffic games添加到keywords的第一位"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # 查找meta keywords标签
        pattern = r'<meta name="keywords" content="([^"]+)"'
        match = re.search(pattern, content)
        
        if match:
            current_keywords = match.group(1)
            # 检查是否已经有traffic games
            if not current_keywords.lower().startswith('traffic games'):
                # 在开头添加traffic games
                new_keywords = f"traffic games, {current_keywords}"
                new_content = content.replace(
                    f'<meta name="keywords" content="{current_keywords}"',
                    f'<meta name="keywords" content="{new_keywords}"'
                )
                
                with open(file_path, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"✓ Fixed: {file_path}")
                return True
            else:
                print(f"  Skipped (already has traffic games first): {file_path}")
                return False
        return False
    except Exception as e:
        print(f"✗ Error: {file_path} - {e}")
        return False

def main():
    base_dir = r"f:\DATA\MyWorkspace\h5game\traffic games"
    categories = ['clicker', 'driving', 'escape', 'parking', 'racing', 'trafficControl']
    
    fixed_count = 0
    
    for category in categories:
        category_path = os.path.join(base_dir, category)
        if os.path.exists(category_path):
            for filename in os.listdir(category_path):
                if filename.endswith('.html'):
                    file_path = os.path.join(category_path, filename)
                    if fix_keywords_in_file(file_path):
                        fixed_count += 1
    
    print(f"\nTotal files fixed: {fixed_count}")

if __name__ == "__main__":
    main()

File: fix_keywords.js
const fs = require('fs');
const path = require('path');

const categories = ['clicker', 'driving', 'escape', 'parking', 'racing', 'trafficControl'];
const baseDir = 'f:\\DATA\\MyWorkspace\\h5game\\traffic games';
let fixedCount = 0;

categories.forEach(category => {
    const categoryPath = path.join(baseDir, category);
    if (fs.existsSync(categoryPath)) {
        const files = fs.readdirSync(categoryPath).filter(f => f.endsWith('.html'));
        files.forEach(file => {
            const filePath = path.join(categoryPath, file);
            let content = fs.readFileSync(filePath, 'utf8');
            
            const match = content.match(/<meta name="keywords" content="([^"]+)"/);
            if (match) {
                const currentKeywords = match[1];
                if (!currentKeywords.toLowerCase().startsWith('traffic games')) {
                    const newKeywords = `traffic games, ${currentKeywords}`;
                    const newContent = content.replace(
                        `<meta name="keywords" content="${currentKeywords}"`,
                        `<meta name="keywords" content="${newKeywords}"`
                    );
                    fs.writeFileSync(filePath, newContent, 'utf8');
                    console.log(`Fixed: ${filePath}`);
                    fixedCount++;
                } else {
                    console.log(`Skipped: ${filePath}`);
                }
            }
        });
    }
});

console.log('');
console.log(`Total files fixed: ${fixedCount}`);
